spec/controllers/api/teams_controller_spec.rb

```
@@ -5,82 +5,156 @@
 
     it 'should get all teams for no query' do
-      login_as(:alice)
+      login_as(:bob)
+
       get :index, params: { 'q': '' }, xhr: true
 
+      expect(data.size).to eq(2)
+
       result_json = data.first
 
-      team = teams(:team1)
+      expect(result_json['attributes']['name']).to eq team1.name
+      expect(result_json['id']).to eq team1.id.to_s
 
-      expect(result_json['attributes']['name']).to eq team.name
-      expect(result_json['id']).to eq team.id.to_s
+      result_json = data.second
+
+      expect(result_json['attributes']['name']).to eq team2.name
+      expect(result_json['id']).to eq team2.id.to_s
     end
 
     it 'should get a single team if one team_id is given' do
-      login_as(:alice)
-
-      team = teams(:team1)
+      login_as(:bob)
 
-      get :index, params: { 'team_ids': [team.id] }, xhr: true
+      get :index, params: { 'team_id': team1.id }, xhr: true
 
-      expect(data.count).to eq(1)
+      expect(data).not_to be(Array)
       expect(response.status).to be(200)
 
-      attributes = data.first['attributes']
-
-      expect(data.second).to be(nil)
+      attributes = data['attributes']
 
       included_types = json['included'].map { |e| e['type'] }
 
       expect(included_types).to include('folder'.pluralize)
       expect(included_types).to include('account'.pluralize)
 
-      expect(attributes['name']).to eq team.name
-      expect(attributes['description']).to eq team.description
+      expect(attributes['name']).to eq team1.name
+      expect(attributes['description']).to eq team1.description
+
+      expect(attributes['name']).not_to eq team3.name
+      expect(attributes['description']).not_to eq team3.description
     end
 
-    it 'should get multiple teams if more than one id is given' do
+    it 'should not get team if not member' do
+      login_as(:alice)
+
+      expect do
+        get :index, params: { 'team_id': team2.id }, xhr: true
+      end.to raise_error(ActiveRecord::RecordNotFound)
```

very important test case !
this should return a 401 error ! use pundit scope in controller !